### PR TITLE
Bugfix/adding function to mark user completed

### PR DESCRIPTION
**Pull request by vivek-kumar-pandey**
### Types of changes

- [x] Bug fix (a change which fixes an issue)
- [ ] New feature (a change which adds functionality)

### Short description of what this resolves:

The functionality to mark users as completed after their program completes was missing so adding them back.

### Checklist:

- [x] I have performed a self-review of my own code.
- [x] The code follows the style guidelines of this project.
- [ ] The code changes are passing the CI checks
- [ ] I have documented my code wherever required.
- [ ] The changes requires a change to the documentation.
- [ ] I have updated the documentation based on the my changes.

### CODE REVIEW

File: api/services/user_program_service.py [hunk 1]
```diff
@@ -15,3 +13,12 @@ def set_init_data(self, jsonData):
...
         user = models.User.query.get_by_phone(self.user_phone)
         self.user_id = user.id
+
+    def mark_user_program_as_completed(self, JsonData):
+        self.set_init_data(JsonData)
+        self.user_program_data = models.UserProgram.get_by_user_id(self.user_id)
+        if self.user_program_data:
+            self.user_program_data.status = (
+                models.UserProgram.UserProgramStatus.COMPLETE
+            )
```
Comment (Satendra-SR) [inline api/services/user_program_service.py:23]: While marking the user_program status as **completed** we need to update the end_date of the user_program also.

**Review by Satendra-SR — CHANGES_REQUESTED** [review]
One suggestion else looks good.

**Review by GauravGusain98 — APPROVED** [review]
LGTM

**Review by Satendra-SR — APPROVED** [review]
Looks fine.

**Pull request merged**